.github/workflows/doxygen.yml

```diff
@@ -0,0 +1,38 @@
+name: Documentation Builder
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - CHANGELOG.md
+
+jobs:
+  Doxygen:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+      - name: Setup
+        run: |
+          sudo apt update && sudo apt install -y doxygen graphviz
+          cp include/aws/cryptosdk/version.h.in include/aws/cryptosdk/version.h
+          source versions
+          SOURCE_DIR=$(pwd) doxygen doxygen/doxygen.config
+      - name: Commit
+        run: |
+          if [ ! -d "doxygen/html" ]; then exit 255; fi
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          git fetch origin
+          git checkout gh-pages
+          rm -rf html
+          mv doxygen/html .
+          git add --all
+          git commit -m "GHA Doxygen"
```

Comment:
Can/should you `git commit -a`?  That will combine the line above...

Reply:
It may be that `git add html/*` would be a better way.

Reply:
dOxygen creates a static website with a few different sub-dirs, I don't want to accidentally leave any behind.